.github/workflows/main.yaml

```diff
@@ -0,0 +1,73 @@
+name: Main CI
+
+on: [push]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        python-version:
+          - '3.12'
+    steps:
+    - uses: actions/checkout@v4
+    - name: Use Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+        cache: 'pip'
+    - name: Install testing dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements-test.txt
```

Comment:
I thought `requrements.txt` already contained test dependencies, so could we just add the new packages to that file perhaps?

Reply:
`flake8` isn't in `requirements.txt`, so I think this is till needed for the additional test code below.

Reply:
The intent of `requiremenst-test.txt` was to avoid those deps being installed other than for testing.  No need to install a linter for a regular install.  What is best practice for that concept?  I saw that somewhere and it seemed a reasonable approach.